generators/base/logic/output/renderer.mjs

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2013-2022 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import path from 'path';
+import ejs from 'ejs';
+// eslint-disable-next-line  import/no-named-default
+import { appendYeomanOptionsFromGeneratorOptions } from '../options.mjs';
+import { resetFakerSeed } from '../sequences.mjs';
+
+const appendCallBackToTemplate = (generator, cb, promise, source) => {
+  if (cb) {
+    return promise
+      .then(res => cb(res))
+      .catch(err => {
+        generator.warning(`Copying template ${source} failed. [${err}]`);
+        throw err;
+      });
+  }
+  return promise;
+};
```

Comment:
??? why?
- it uses promises, await/async is much cleaner.

Reply:
I just extracted the code as-is.

Reply:
seems that this callback is always needed by many methods, leaving as-is for now